pil-analyzer/src/expressionizer.rs

```diff
@@ -59,6 +59,9 @@ pub fn try_value_to_expression<T: FieldElement>(
 }
 
 struct Expressionizer<'a> {
+    /// Maps polynomial IDs to their names.
+    /// For arrays, this does not include the array elements, just the ID
+    /// of the first element (the array itself).
```

Comment:
```suggestion
    /// Maps polynomial IDs to their names.
    /// For arrays, this does not include the array elements, just the ID
    /// of the first element (the array itself).
    /// Crucially, the polynomial type comes first, so that we can still find
    /// the array's name given an element's ID by using the next smaller entry.
```